src/compiler/gcc.rs

```diff
@@ -169,6 +169,7 @@ fn _parse_arguments(arguments: &[OsString],
                 v if v.starts_with('@') => return CompilerArguments::CannotCache("@file"),
                 "-M" |
                 "-MM" |
+                "-MP" |
```

Comment:
You should mention in the commit message that this not being recognized as a preprocessor flag made gcc error out without preprocessor flags because gcc wants other -M flags alongside.